addons/bus/websocket.py

```diff
@@ -631,6 +636,19 @@ def _trigger_lifecycle_event(self, event_type):
                         exc_info=True
                     )
 
+    def _trim_sent_notifs(self):
+        """
+        Clears old sent notifications and updates the minimum ID to fetch.
+
+        Since smaller ID notifications might be committed before larger ones,
+        fetching notifications with IDs only greater than the last sent ID could
+        lead to missing some. To prevent this, we track notifications sent in
+        the past 10 seconds to ensure none are missed.
+        """
+        last_sent_id = max((notif[0] for notif in self._sent_notifs), default=self._min_notif_id)
+        self._sent_notifs = [notif for notif in self._sent_notifs if time.time() - notif[1] < 10]
+        self._min_notif_id = min(notif[0] for notif in self._sent_notifs, default=last_sent_id)
```

Comment:
I think you had that and reverted it at some point but it should be kept
```suggestion
        self._min_notif_id = min((notif[0] for notif in self._sent_notifs), default=last_sent_id)
```